blocks/core/_lib/ff-tabs/_ff-tabs-core.js

```diff
@@ -113,16 +122,61 @@ function getSwapStatesMethod($root, options) {
                 addCompleteClasses($lastLinks, $lastContent);
             }
         }
+    }
+
+    function handleClick(e) {
+        e.preventDefault();
+        /*jshint validthis:true */
+        setState(this);
+    }
 
-    };
-}
 
-function setClickHandler($root, options) {
-    var linkSel = '[' + options.linkSelBase + ']';
-    $root.on('click', linkSel, getSwapStatesMethod($root, options));
+    function testBounds(value, length) {
+        if (value < 0) {
+            return length - 1;
+        } else if (value >= length) {
+            return 0;
+        } else return value;
+    }
+
+    function setActiveTab(index) {
```

Comment:
I tend to only add (and test) things that are required by production code, but I don't think I'm really in a position to comment on how you're developing your code! I was just thinking more about what gets exposed to the user -- it's easier to expose more things, but harder to take them away/change them.